homeassistant/components/denonavr/media_player.py

```diff
@@ -310,77 +369,92 @@ def media_episode(self):
-        return self._receiver.next_track()
+        await self._receiver.async_next_track()
 
-    def select_source(self, source):
+    @async_log_errors
+    async def async_select_source(self, source: str):
         """Select input source."""
         # Ensure that the AVR is turned on, which is necessary for input
         # switch to work.
-        self.turn_on()
-        return self._receiver.set_input_func(source)
+        await self.async_turn_on()
+        await self._receiver.async_set_input_func(source)
 
-    def select_sound_mode(self, sound_mode):
+    @async_log_errors
+    async def async_select_sound_mode(self, sound_mode: str):
         """Select sound mode."""
-        return self._receiver.set_sound_mode(sound_mode)
+        await self._receiver.async_set_sound_mode(sound_mode)
 
-    def turn_on(self):
+    @async_log_errors
+    async def async_turn_on(self):
         """Turn on media player."""
-        if self._receiver.power_on():
-            self._state = STATE_ON
+        await self._receiver.async_power_on()
 
-    def turn_off(self):
+    @async_log_errors
+    async def async_turn_off(self):
         """Turn off media player."""
-        if self._receiver.power_off():
-            self._state = STATE_OFF
+        await self._receiver.async_power_off()
 
-    def volume_up(self):
+    @async_log_errors
+    async def async_volume_up(self):
         """Volume up the media player."""
-        return self._receiver.volume_up()
+        await self._receiver.async_volume_up()
 
-    def volume_down(self):
+    @async_log_errors
+    async def async_volume_down(self):
         """Volume down media player."""
-        return self._receiver.volume_down()
+        await self._receiver.async_volume_down()
 
-    def set_volume_level(self, volume):
+    @async_log_errors
+    async def async_set_volume_level(self, volume: int):
         """Set volume level, range 0..1."""
         # Volume has to be sent in a format like -50.0. Minimum is -80.0,
         # maximum is 18.0
         volume_denon = float((volume * 100) - 80)
         if volume_denon > 18:
             volume_denon = float(18)
-        with suppress(ValueError):
-            if self._receiver.set_volume(volume_denon):
-                self._volume = volume_denon
+        await self._receiver.async_set_volume(volume_denon)
 
-    def mute_volume(self, mute):
+    @async_log_errors
+    async def async_mute_volume(self, mute: bool):
         """Send mute command."""
-        return self._receiver.mute(mute)
+        await self._receiver.async_mute(mute)
 
-    def get_command(self, command, **kwargs):
+    @async_log_errors
+    async def async_get_command(self, command: str, **kwargs):
         """Send generic command."""
-        self._receiver.send_get_command(command)
+        return await self._receiver.async_get_command(command)
+
+    # Decorator defined before is a staticmethod
+    async_log_erros = staticmethod(  # pylint: disable=no-staticmethod-decorator
```

Comment:
`async_log_erros`?

Reply:
omg, I missed that 😅 
I'm not happy with my linter settings in vscode + devcontainers yet. There are too many pycodestyle `line too long` errors.